### PR TITLE
Review comments

### LINKED ISSUE #1: Add license headers

### LINKED ISSUE #2: Add header for constants
A header that can be used by consumers defining constants, i.e. `SignatureAlgorithm`, should be added.

### PR DESCRIPTION
**Pull request by franziskuskiefer**
Addresses Tim's review comments and added a header file for convenience.
fixes #1, fixes #2.

### CODE REVIEW

File: include/cosec.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <stdint.h>
+
+enum CoseSignatureType { ES256 = 0, ES384 = 1, ES512 = 2, PS256 = 3 };
+
+extern "C" {
+typedef bool (*cose_verify_callback)(const uint8_t *payload, size_t payload_len,
```
Comment (mozkeeler) [inline include/cosec.h]: '*' to the left?

File: src/lib.rs [hunk 2]
```diff
@@ -39,17 +43,17 @@ pub extern "C" fn verify_cose_signature_ffi(
...
         Err(_) => Vec::new(),
     };
-    if cose_signatures.len() < 1 {
+    if cose_signatures.len() == 0 {
         return false;
     }
 
-    let mut result = true;
-    for cose_signature in cose_signatures {
+    // for cose_signature in cose_signatures {
```
Comment (mozkeeler) [inline src/lib.rs]: If this is a comment explaining the following code, let's comment in prose. If it's just commented-out code, let's remove it.

File: include/cosec.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                                     const uint8_t **cert_chain,
+                                     size_t cert_chain_len,
+                                     const size_t *certs_len,
+                                     const uint8_t *ee_cert, size_t ee_cert_len,
+                                     const uint8_t *signature,
+                                     size_t signature_len, uint8_t algorithm);
+bool verify_cose_signature_ffi(const uint8_t *payload, size_t payload_len,
```
Comment (mozkeeler) [inline include/cosec.h]: It occurs to me that at some point we may want to have an automated test that ensures that the rust declaration of this API matches the C declaration...

**Review by mozkeeler — APPROVED** [review]
Cool - just a few comments.

File: include/cosec.h (hunk 3 above)
Comment (franziskuskiefer) [reply include/cosec.h]: Yeah that would be great. Not sure if there's something like that already. It sounds to me like something that could be interesting to others as well.

File: src/lib.rs (hunk 2 above)
Comment (franziskuskiefer) [reply src/lib.rs]: This was me being stupid again.

**Pull request merged**